### PR TITLE
Consolidated C# operators overviews

### LINKED ISSUE #6981: Missing Associativity
Missing Associativity

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7eeac581-6235-2192-d909-e7fe3d21197f
* Version Independent ID: 9d8e2a57-1c18-8cba-9bd4-eef546fe2dcb
* Content: [C# Operators](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/index)
* Content Source: [docs/csharp/language-reference/operators/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/operators/index.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by pkulikov**
Merged and revised the overview from the programming guide into the one from the language reference.

Fixes #6981

### CODE REVIEW

File: docs/csharp/language-reference/operators/index.md [hunk 1]
```diff
@@ -45,7 +52,39 @@ The following table lists the C# operators starting with the highest precedence
...
+Unrelated to operator precedence and associativity, operands in an expression are evaluated from left to right. The following examples demonstrate the order in which operators and operands are evaluated:
+
+| Expression | Order of evaluation |
+| ---------- | ------------------- |
+|`a + b`|a, b, +|
+|`a + b * c`|a, b, c, *, +|
+|`a / b + c * d`|a, b, /, c, d, *, +|
+|`a / (b + c) * d`|a, b, c, +, /, d, *|
+
+Typically, all operator operands are evaluated. The operators that can skip evaluation of some of its operands are the conditional logical [AND (`&&`)](boolean-logical-operators.md#conditional-logical-and-operator-) and [OR (`||`)](boolean-logical-operators.md#conditional-logical-or-operator-) operators, the [null-coalescing operator `??`](null-coalescing-operator.md), the [null-conditional operators `?.` and `?[]`](member-access-operators.md#null-conditional-operators--and-), and the [conditional operator `?:`](conditional-operator.md).
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/index.md]: I think a very brief explanation about why these operators conditionally evaluate the right operand would be useful. Something like this:

```suggestion
Typically, all operator operands are evaluated. Some operators evaluate operands conditionally, where later operands are only evaluated when the first operand satisfies some condition. These operators are the conditional logical [AND (`&&`)](boolean-logical-operators.md#conditional-logical-and-operator-) and [OR (`||`)](boolean-logical-operators.md#conditional-logical-or-operator-) operators, the [null-coalescing operator `??`](null-coalescing-operator.md), the [null-conditional operators `?.` and `?[]`](member-access-operators.md#null-conditional-operators--and-), and the [conditional operator `?:`](conditional-operator.md). See the description of each operator for more details.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks again for making these changes @pkulikov I like how you've brought the information together.

I had one concern, and a possible suggestion. Let me know what you think, or if you have alternate wording.

Thanks again.

File: docs/csharp/language-reference/operators/index.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/language-reference/operators/index.md]: @BillWagner I agree with you: conditional evaluation is what those operators share, so worth a mention. I've used alternative wording. Please take a look at the latest commit.

Comment (BillWagner) [reply docs/csharp/language-reference/operators/index.md]: I like that wording. It describes why the behavior is different and does it very well.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks again @pkulikov 

This is ready to :shipit:.

**Pull request merged**